lib/active_merchant/billing/gateways/deepstack.rb

```diff
@@ -0,0 +1,388 @@
+        post[:payment_instrument][:credit_card][:account_number] = creditcard.number
+        post[:payment_instrument][:credit_card][:expiration] = '%02d%02d' % [creditcard.month, creditcard.year % 100]
+        post[:payment_instrument][:credit_card][:cvv] = creditcard.verification_value
+      end
+
+      def add_order(post, amount, options)
+        post[:transaction] = {} unless post.key?(:transaction)
+
+        post[:transaction][:amount] = amount
+        post[:transaction][:cof_type] = options.key?(:cof_type) ? options[:cof_type].upcase : 'UNSCHEDULED_CARDHOLDER'
+        post[:transaction][:capture] = false # Change this in the request (auth/charge)
+        post[:transaction][:currency_code] = (options[:currency] || currency(amount).upcase)
+        post[:transaction][:avs] = options[:avs] || true # default avs to true unless told otherwise
+        post[:transaction][:save_payment_instrument] = options[:save_payment_instrument] || false
+      end
+
+      def parse(body)
+        return {} if !body || body.empty?
+
+        JSON.parse(body)
+      end
+
+      def commit(action, parameters, method = 'POST')
+        url = (test? ? test_url : live_url)
+        if no_hmac(action)
+          request_headers = headers.merge(create_basic(parameters, action))
+        else
+          request_headers = headers.merge(create_hmac(parameters, method))
+        end
+        request_url = url + get_url(action)
+        begin
+          response = parse(ssl_post(request_url, post_data(action, parameters), request_headers))
+          Response.new(
+            success_from(response),
+            message_from(response),
+            response,
+            authorization: authorization_from(response),
+            avs_result: AVSResult.new(code: response['avs_result']),
+            cvv_result: CVVResult.new(response['cvv_result']),
+            test: test?,
+            error_code: error_code_from(response)
+            # authorization: authorization_from(response)
+          )
+        rescue ResponseError => e
+          Response.new(
+            false,
+            message_from_error(e.response.body),
+            response_error(e.response.body)
+          )
+        rescue JSON::ParserError
+          Response.new(
+            false,
+            message_from(response),
+            json_error(response)
+          )
+        end
+      end
+
+      def headers
+        {
+          'Accept' => 'text/plain',
+          'Content-Type' => 'application/json'
+        }
+      end
+
+      def response_error(response)
+        parse(response)
+      rescue JSON::ParserError
+        json_error(response)
+      end
+
+      def json_error(response)
+        msg = 'Invalid response received from the Conekta API.'
+        msg += "  (The raw response returned by the API was #{response.inspect})"
+        {
+          'message' => msg
+        }
+      end
+
+      def success_from(response)
+        # begin
```

Comment:
Same, left commented code here. There are other commented code lines left in the tests as well